### DISCUSSION TITLE
podman save question - size differences between formats docker-archive and oci-archive?

### DISCUSSION THREAD

**Discussion by Jared-Sprague** [containers/podman#17934] [Q&A] [answered]
Hello!

I'm working with saving a large image, and I noticed that when running `podman save` with the default `docker-archive` format the image is MUCH larger than when I use `--format oci-archive`

Example:
`podman save -o myimage-docker-archive.tar myimage`
`du -hs myimage-docker-archive.tar # 14G`
`podman save --format oci-archive -o myimage-oci-archive.tar myimage`
`du -hs myimage-oci-archive.tar # 7.4G`

The `oci-archive` was 6.6G smaller!!!  

First: go oci-archive format, that's amazing.
Second: why the size difference?

Is oci-archive employing compression during the save process on certain files?  If so, what files, and what compression format is is using?  I noticed that it compresses even more than gzip -9

Thanks!

**Comment by vrothberg** [comment]
Thanks for reaching out @Jared-Sprague!

The layers inside the oci-archive are compressed while the layers inside the docker-archive are not compressed.  That can make quite a difference.

@mtrmac anything you want to add?

**Comment by mtrmac** [comment, marked as answer]
Yes, the default of `oci-archive` is compression; that’s just gzip with no special options (but a not quite default implementation), AFAIK.

`docker-archive` layers are always uncompressed (but it is customary to compress the output afterwards).

---

Just by a quick grep:

c/image’s `types.SystemContext.OCIAcceptUncompressedLayers` seems to be provided by libimage, and visible as `podman save --uncompressed`

So it should be possible to set the two on a level footing.

**Comment by Jared-Sprague** [comment]
Thanks!
